design/annotation-for-power-cycling-and-deleting-failed-nodes.md

```diff
@@ -115,45 +115,41 @@ recover exclusive workloads and restore cluster capacity.
 
 ### Implementation Details/Notes/Constraints [optional]
 
-The controller includes logic for power cycling a node.  
-
-As rebooting (software) and power cycling (hardware) a machine is a long
-running multi-step process, there are currently discussions around the creation
-of a public mechanism for requesting initiating these processes atomically.
-
-While it is likely that the work proposed here may make use of such
-functionality once it exists, it should not be considered a pre-requisite for
-the purposes of evaluating this proposal.
+The controller shall be called machine-remediator-baremetal
+The annotation to trigger remediation is `machine.openshift.io/external-remediation`
```

Comment:
What is the content of this annotation?

Reply:
Just an empty string:

  `t.Machine.Annotations[machineRebootAnnotationKey] = ""`

Reply:
It'd be nice if it had some unique value (timestamp, UID) that we could then re-use in the BareMetalHost reboot to make sure we don't get stuck in a loop.

Reply:
It was a non issue on the Node because the annotation was wiped when we deleted the Node. I’ll have a think about what loops we might be able to trigger

Reply:
If the annotation triggering all of this is still on the node, it still will be deleted when the node is deleted, right?

Reply:
It's on the Machine now.
I don't agree that this was a non-issue before; the code that was written to ensure that we only do one reboot doesn't work, and to make it work we'd need to use a consistent name that could only come from the content of the annotation. We'd have to do exactly the same thing with a RebootRequest resource. With this annotation we will want to do something equivalent, otherwise we don't know if an existing annotation on the baremetalhost was placed there by us or by somebody else.